src/branchPort.ts

```diff
@@ -128,11 +131,47 @@ export default class BranchPort {
       for (const label of labels.data) {
         const parts = label.name.split('port:');
         if (parts.length === 2 && parts[1]) {
-          this.targetBranches.push(parts[1]);
+          this.targetBranches.push(await this.getLatestReleaseBranch(parts[1]));
         }
       }
     }
 
     return this.targetBranches;
   }
+
+  /*
+   * If the label ends with '-latest', substitute the target with the
+   * latest release branch.
+   *
+   * Example format: release/v-latest
+   * Where 'release/v' represents the initial prefix of the release branch.
+   * We then check that the standard release format follows the prefix
+   * (\d+.\d+.\d+ or \d+-\d+-\d+).
+   */
+  public async getLatestReleaseBranch(target: string): Promise<string> {
+    var newTarget = target;
+    if (target.endsWith('-latest')) {
+      var releaseBranches = [];
+      const releaseBranchPrefix = target.split('-latest')[0];
+      const myRepoID = this.context.repo({
+        number: this.context.payload.repository.id
+      });
+      var branches = await this.context.github.repos.listBranches(myRepoID);
+      
+      for (var branch of branches.data) {
+        if (branch.name.startsWith(releaseBranchPrefix) && 
+            branch.name.split(releaseBranchPrefix)[1].match(/^([\d]+[\.-]+)+[\d]+$/)
+        ) {
+          releaseBranches.push(branch.name);
+        }
+      }
+      if (releaseBranches.length == 0) {
+        throw new Error(
+          'No release branches found, but release label was added.'
+        );
```

Comment:
```suggestion
        const e = new Error(
          'No release branches found, but release label was added.'
        );
        e.name = 'ReleaseBranchNotFound';
        e.targetBranchName = target;
        throw e;
```

This error is thrown when a call to `getTargetBranches` is made, which only happens [here](https://github.com/brpowell/autointegrator/pull/27/files#diff-f41e9d04a45c83f3b6f6e630f10117feR37). That line needs to be included in the try block in addition to this update. [This](https://github.com/brpowell/autointegrator/pull/27/files#diff-68d09c8c97d4d7d6f61632c11a99f9c2R76) also needs to be removed since that try block never calls the method and won't catch that case.

Reply:
I can't quite see what your links are too (they forward to the same place for me). I do see what you mean though. Originally I had called the release branch logic inside of the try catch, and later thought it would fit better tied to getTargetBranches. I didn't update the error handling - thanks for this catch!